### PR TITLE
fixing more pagination in research notes

### PR DESCRIPTION
**Pull request by max-keviv**
As discussed in issue #10219
I move the line 132-133 after 143

### CODE REVIEW

**Review by TildaDares — CHANGES_REQUESTED** [review]

File: app/controllers/tag_controller.rb [hunk 1]
```diff
@@ -141,6 +138,9 @@ def show
       @notes = nodes.where('node.nid NOT IN (?)', @qids) if @node_type == 'note'
       @questions = nodes.where('node.nid IN (?)', @qids) if @node_type == 'questions'
     end
+    
+    @pagy, nodes = pagy(nodes.order(order_by), items: 24)
+    @paginated = true
```
[the remark this answers is not in the document]
Comment (max-keviv) [reply app/controllers/tag_controller.rb:143]: so @paginated on line 143 is right or it should also be needed to move

[the remark this answers is not in the document]
Comment (TildaDares) [reply app/controllers/tag_controller.rb:143]: The only thing that needs to change is `nodes` being sorted before line 138. The rest can stay.

File: app/controllers/tag_controller.rb [hunk 2]
```diff
@@ -141,6 +138,9 @@ def show
       @notes = nodes.where('node.nid NOT IN (?)', @qids) if @node_type == 'note'
       @questions = nodes.where('node.nid IN (?)', @qids) if @node_type == 'questions'
     end
+    
+    @pagy, nodes = pagy(nodes.order(order_by), items: 24)
```
Comment (jywarren) [inline app/controllers/tag_controller.rb:142]: So I think we need to make a new line on 137:

`nodes = nodes.order(order_by)`

Then we can change this line to:

```suggestion
    @pagy, nodes = pagy(nodes, items: 24)
```

That way the sorting still happens above, but the pagination happens below. Sound good?

File: app/controllers/tag_controller.rb [hunk 3]
```diff
@@ -129,9 +129,6 @@ def show
                  'node_revisions.timestamp DESC'
                end
 
-    @pagy, nodes = pagy(nodes.order(order_by), items: 24)
-    @paginated = true
-
     @qids = Node.questions.where(status: 1)
```
Comment (jywarren) [inline app/controllers/tag_controller.rb:132]:
```suggestion
    nodes = nodes.order(order_by)
    @qids = Node.questions.where(status: 1)
```

**Comment by jywarren** [comment]
I added two suggestions you can accept. @TildaDares feel free to merge this once those are complete!

File: app/controllers/tag_controller.rb (hunk 3 above)
Comment (max-keviv) [reply app/controllers/tag_controller.rb:132]: @jywarren  thank you for suggestions

**Comment by max-keviv** [comment]
open pull request at #10588

**Pull request closed without merging**